File: backend/src/Core.Query/Queries/User/UserAuctions/UserAuctionsQuery.cs
﻿using System;
using System.ComponentModel.DataAnnotations;
using Core.Common;
using Core.Common.Attributes;
using Core.Common.Domain.Users;
using Core.Common.Query;
using MediatR;

namespace Core.Query.Queries.User.UserAuctions
{
    public enum UserAuctionsSorting
    {
        DATE_CREATED = 0
    }

    public enum UserAuctionsSortDir
    {
        DESCENDING = 0, ASCENDING
    }

    [AuthorizationRequired]
    public class UserAuctionsQuery : IQuery<UserAuctionsQueryResult>
    {
        [Range(0, Int32.MaxValue)]
        public int Page { get; set; } = 0;

        public bool ShowArchived { get; set; } = false;

        public UserAuctionsSorting Sorting { get; set; }

        public UserAuctionsSortDir SortingDirection { get; set; }

        [SignedInUser]
        public UserIdentity SignedInUser { get; set; }
    }
}

File: front/src/app/core/queries/UserAuctionsQuery.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Auction } from '../models/Auctions';
import { QueryHelper } from './QueryHelper';
import { environment } from '../../../environments/environment';

export enum UserAuctionsSorting{
  DATE_CREATED = 0
}

export enum UserAuctionsSortDir{
  DESCENDING = 0, ASCENDING
}

export interface UserAuctions {
  auctions: Auction[];
  total: number;
}

@Injectable({
  providedIn: 'root'
})
export class UserAuctionsQuery {
  constructor(private httpClient: HttpClient, private queryHelper: QueryHelper) {
  }

  execute(pageNum: number, showArchived: boolean, sorting: UserAuctionsSorting, dir: UserAuctionsSortDir): Observable<UserAuctions> {
    const url = `${environment.API_URL}/api/q/userAuctions?page=${pageNum}&show-archived=${showArchived}&sort=${sorting}&dir=${dir}`;
    let req = this.httpClient.get<UserAuctions>(url);
    return this.queryHelper.pipeLoading(req);
  }
}
